### PR TITLE
Add webserverConfigConfigMapName

### PR DESCRIPTION
**Pull request by JulesTriomphe**
This PR adds a webserverConfigConfigmapName to use a webserver_config.py in a custom configmap.
Specifying a value for this configuration overrides the webserverConfig configuration.

This PR is also designed to solve the issue with readOnlyRootFilesystem clusters, and hence mounts the configmap as a volume for all appropriate containers.

### CODE REVIEW

**Comment by JulesTriomphe** [comment]
Hi @dstandish and @jedcunningham, this PR is also ready for review

**Comment by JulesTriomphe** [comment]
The failing check runs fine for me locally. I don't get why it fails here...

**Comment by jedcunningham** [comment]
Let's just rerun it as a first attempt, as it seems like a time based failure.

**Comment by JulesTriomphe** [comment]
Looks like it worked. Thanks @jedcunningham !

File: chart/values.yaml [hunk 1]
```diff
@@ -976,6 +978,7 @@ webserver:
 
   #   # Flask-WTF flag for CSRF
   #   CSRF_ENABLED = True
+  webserverConfigConfigmapName: ~
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
  webserverConfigConfigMapName: ~
```
We should match the casing used by k8s.

File: tests/charts/test_webserver.py [hunk 2]
```diff
@@ -738,6 +750,45 @@ def test_webserver_config_configmap(self):
             == jmespath.search('data."webserver_config.py"', docs[0]).strip()
         )
 
+    def test_webserver_config_configmap_name_volume_mounts(self):
+        configmap_name = "my-configmap"
+        docs = render_chart(
+            values={
+                "scheduler": {"logGroomerSidecar": {"enabled": True}, "waitForMigrations": {"enabled": True}},
+                "triggerer": {"waitForMigrations": {"enabled": True}},
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py]:
```suggestion
```

We don't need to set these defaults.

File: tests/charts/test_webserver.py [hunk 3]
```diff
@@ -738,6 +750,45 @@ def test_webserver_config_configmap(self):
...
+                "webserver": {
+                    "waitForMigrations": {"enabled": True},
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py]:
```suggestion
```

Same here.

File: tests/charts/test_webserver.py [hunk 4]
```diff
@@ -738,6 +750,45 @@ def test_webserver_config_configmap(self):
...
+                    "webserverConfig": "CSRF_ENABLED = True  # {{ .Release.Name }}",
+                    "webserverConfigConfigmapName": configmap_name,
+                },
+                "workers": {"kerberosSidecar": {"enabled": True}, "persistence": {"enabled": True}},
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py]:
```suggestion
                "workers": {"kerberosSidecar": {"enabled": True}},
```

And here.

File: tests/charts/test_webserver.py [hunk 5]
```diff
@@ -738,6 +750,45 @@ def test_webserver_config_configmap(self):
...
+            },
+            show_only=[
+                "templates/scheduler/scheduler-deployment.yaml",
+                "templates/triggerer/triggerer-deployment.yaml",
+                "templates/webserver/webserver-deployment.yaml",
+                "templates/workers/worker-deployment.yaml",
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py]: This test should probably be moved to `test_airflow_common.py` instead.

File: tests/charts/test_webserver.py (hunk 2 above)
Comment (JulesTriomphe) [reply tests/charts/test_webserver.py]: Without these we don't test whether the configMap has been mounted to these initContainers and sidecars. I think we should leave these defaults.

File: tests/charts/test_webserver.py (hunk 3 above)
Comment (JulesTriomphe) [reply tests/charts/test_webserver.py]: Same here

File: tests/charts/test_webserver.py (hunk 4 above)
Comment (JulesTriomphe) [reply tests/charts/test_webserver.py]: And here for the worker-log-groomer

File: tests/charts/test_webserver.py (hunk 2 above)
Comment (jedcunningham) [reply tests/charts/test_webserver.py]: Sorry, let me be more clear. All of these are already true by default, you don't need to set them explicitly here, and they will still be tested.

e.g. [scheduler.logGroomerSidecar.enabled](https://github.com/apache/airflow/blob/19f1e5bf4b9544ae9775ebd8c57797335be53ed6/chart/values.yaml#L722), [scheduler.waitForMigrations.enabled](https://github.com/apache/airflow/blob/19f1e5bf4b9544ae9775ebd8c57797335be53ed6/chart/values.yaml#L739), [triggerer.waitForMigrations.enabled](https://github.com/apache/airflow/blob/19f1e5bf4b9544ae9775ebd8c57797335be53ed6/chart/values.yaml#L1129)

Comment (JulesTriomphe) [reply tests/charts/test_webserver.py]: Ahh yes, ok, thanks for pointing that out

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**